scan/lib/scan/detect_values.rb

```diff
@@ -162,6 +163,7 @@ def self.detect_simulator(devices, requested_os_type, deployment_target_key, def
               filter_simulators(simulators, :equal, version).tap(&potential_emptiness_error).select(&selector)
             end
           ).tap do |array|
+            fail_if_device_not_found(device_string) if Scan.config[:ensure_devices_found] && array.empty?
```

Comment:
Can we put this line and the one below inside the same `if array.empty?`? 🙃  Seems like it may read easier and be easier to maintain in the future 🤷‍♂️

Reply:
That is right, I removed first line. 👍
 I just wanted to avoid additional error message `UI.error("No simulators found that are equal to the version " \
                  "of specifier (#{version}) and greater than or equal to the version " \
                  "of deployment target (#{deployment_target_version})")` if you have specified device with specific iOS version and set `deployment_target_version`. But it works anyway, I tested it again 🚀
                  Thanks 🥇